### DISCUSSION TITLE
How to know, whether layer is fully rendered or not in the browser?

### DISCUSSION THREAD

**Discussion by iamtekson** [Leaflet/Leaflet#8981] [Q&A]
### Checklist

- [X] I've looked at the [documentation](https://leafletjs.com/reference.html) to make sure the behavior isn't documented and expected.
- [X] I'm sure this is an issue with Leaflet, not with my app or other dependencies (Angular, Cordova, React, etc.).
- [X] I've searched through the [current issues](https://github.com/Leaflet/Leaflet/issues) to make sure this hasn't been reported yet.
- [X] I agree to follow the [Code of Conduct](https://github.com/Leaflet/Leaflet/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.

### Steps to reproduce

I have wrote the following line of code,

```js
var latlngs = [[37, -109.05], [41, -109.03], [41, -102.05], [37, -102.04]];
var polygon = L.polygon(latlngs, { color: 'red' }).addTo(map);

map.fitBounds(polygon.getBounds());
```

### Expected behavior

Now I am trying to see test whether my layer is fully rendered or not. I know there is the `load` event for `TileLayer` in order to know the layer rendering status. But I am curious, is there any way to know whether my polygon is rendered fully or not?

I have tried following but it is not working (I know `load` event is not available for the all type of layers):

```js
polygon.on('load', () => {
  console.log('polygon fully rendered')
})
```

### Current behavior

Is there any way?

### Environment

- Leaflet version: latest
- Browser (with version): chrome
- OS/Platform (with version): mac

**Comment by iamtekson** [comment]
In mapbox, we can easily know whether all `layers` are loaded or not by using `idle` event on `map` object. I am looking for the similar solution. Any help?

**Comment by IvanSanchez** [comment]
`L.Path`s (including `L.Polygon`s) should be rendered on the next animation frame. Listen to the `update` event on the appropriate `L.Renderer` instance being used by your map instance.

**Comment by iamtekson** [comment]
Thank you very much @IvanSanchez for the quick answer. I have tried the following code based on your suggestion, but still not working. Am I missing something?

```js
var renderer = new L.canvas(); // I tried with L.svg() too
var latlngs = [[37, -109.05], [41, -109.03], [41, -102.05], [37, -102.04]];
var polygon = L.polygon(latlngs, { color: 'red', renderer: renderer }).addTo(map);

//update event
renderer.on('update', () => {
      console.log('polygon loaded')
      renderer.off('update')
    })
```

I notice that the `update` method on renderer only working after map pan or zoom event. But on the first map render, it didn't triggered. Any help please?

**Comment by iamtekson** [comment]
Thanks @IvanSanchez, after changing the order of code, I got the required answer.

```js
    //update event
    renderer.on("update", () => {
      console.log("polygon loaded");
      renderer.off("update");
    });

var polygon = L.polygon(latlngs, { color: 'red', renderer: renderer }).addTo(map);
```

but just wondering, is there any way to check whether geojson layer is fully rendered or not? I don't see anything in the documentation.
